app/code/Magento/InventoryShippingAdminUi/composer.json

```diff
@@ -0,0 +1,30 @@
+{
+  "name": "magento/module-inventory-shipping-admin-ui",
+  "description": "N/A",
+  "require": {
+    "php": "~7.1.3||~7.2.0",
+    "magento/framework": "*",
+    "magento/module-backend": "*",
+    "magento/module-inventory-api": "*",
+    "magento/module-inventory-configuration-api": "*",
+    "magento/module-inventory-sales-api": "*",
+    "magento/module-inventory-source-selection-api": "*",
+    "magento/module-inventory-shipping": "*",
```

Comment:
you should not have a dependency on module inventory-shipping from inventoryShippingAdminUi

moreover, if you will look why you have the dependency you will find that it comes because of class `\Magento\InventoryShipping\Model\IsMultiSourceMode`

Btw, class IsMultiSourceMode not used anymore except UI as it's used for different UI rendering for single and multi-source modes. So, looks like the right approach is to move it to `InventoryShippingAdminUi` module where it's used.

Btw, maybe it makes sense to rename this class because it confuses with `\Magento\InventoryCatalogApi\Model\IsSingleSourceModeInterface`

But their behavior is different: 
IsSingleSourceModeInterface - checks whether we have exactly one active source available (default source in the current implementation)
IsMultiSourceMode - checks that for specified sales channel (website) we have stock with more than one source assigned